contrib/pkg/deprovision/deprovision.go

```diff
@@ -1,26 +1,164 @@
+	ibmcloudutil "github.com/openshift/hive/contrib/pkg/utils/ibmcloud"
+	nutanixutil "github.com/openshift/hive/contrib/pkg/utils/nutanix"
+	openstackutil "github.com/openshift/hive/contrib/pkg/utils/openstack"
+	vsphereutil "github.com/openshift/hive/contrib/pkg/utils/vsphere"
+	"github.com/openshift/hive/pkg/constants"
+
 	"github.com/spf13/cobra"
 )
 
 // NewDeprovisionCommand is the entrypoint to create the 'deprovision' subcommand
 func NewDeprovisionCommand() *cobra.Command {
 	var credsDir string
+	var mjSecretName string
+	var logLevel string
 	cmd := &cobra.Command{
 		Use:   "deprovision",
 		Short: "Deprovision clusters in supported cloud providers",
+		Long: `Platform subcommands use a legacy code path and are deprecated. \
+To run the generic destroyer, use the --metadata-json-secret-name parameter.`,
 		Run: func(cmd *cobra.Command, args []string) {
-			cmd.Usage()
+			if mjSecretName == "" {
+				cmd.Usage()
+				return
+			}
+
+			// Generic deprovision flow using metadata.json
+			logger, err := utils.NewLogger(logLevel)
+			if err != nil {
+				log.Fatalf("failed to create logger: %s", err)
+			}
+
+			c, err := utils.GetClient("hiveutil-deprovision-generic")
+			if err != nil {
+				logger.WithError(err).Fatal("failed to create kube client")
+			}
+
+			// TODO: Refactor LoadSecretOrDie to avoid this setenv/getenv cycle
+			k := "METADATA_JSON_SECRET_NAME"
+			os.Setenv(k, mjSecretName)
+			mjSecret := utils.LoadSecretOrDie(c, k)
+			if mjSecret == nil {
+				// This should not be reachable -- we should have Fatal()ed in LoadSecretOrDie()
+				logger.WithField("secretName", mjSecretName).Fatal("failed to load metadata.json Secret")
+			}
+
+			mjBytes, ok := mjSecret.Data[constants.MetadataJSONSecretKey]
+			if !ok {
+				logger.Fatalf("metadata.json Secret did not contain %q key", constants.MetadataJSONSecretKey)
+			}
+
+			var metadata *types.ClusterMetadata
+			if err = json.Unmarshal(mjBytes, &metadata); err != nil {
+				logger.WithError(err).Fatal("failed to unmarshal metadata.json")
+			}
+
+			platform := metadata.Platform()
+			if platform == "" {
+				logger.Fatal("no platform configured in metadata.json")
+			}
+
+			// TODO: Make a registry or interface for this
+			var ConfigureCreds func(client.Client)
+			switch platform {
+			case aws.Name:
+				ConfigureCreds = awsutil.ConfigureCreds
+			case azure.Name:
+				ConfigureCreds = azureutil.ConfigureCreds
+			case gcp.Name:
+				ConfigureCreds = gcputil.ConfigureCreds
+			case ibmcloud.Name:
+				ConfigureCreds = ibmcloudutil.ConfigureCreds
+			case nutanix.Name:
+				// Snowflake! We need to inject the creds into the metadata.
+				// If env vars are unset, the destroyer will fail organically.
+				ConfigureCreds = func(c client.Client) {
+					nutanixutil.ConfigureCreds(c)
+					metadata.Nutanix.Username = os.Getenv(constants.NutanixUsernameEnvVar)
+					metadata.Nutanix.Password = os.Getenv(constants.NutanixPasswordEnvVar)
+				}
```

Comment:
```suggestion
				// If env vars are unset, the destroyer will fail organically.
				ConfigureCreds = nutanixutil.ConfigureCreds
				metadata.Nutanix.Username = os.Getenv(constants.NutanixUsernameEnvVar)
				metadata.Nutanix.Password = os.Getenv(constants.NutanixPasswordEnvVar)
```

Reply:
I think you can just write it like this.  The function is immediately invoked right below

Reply:
I think adding a metadata parameter and using a nil check to all of the ConfigureCreds implementations is a good compromise, especially if the line count is low (probably no more than +10 per file)